app/models/merchant.rb

```diff
@@ -3,14 +3,16 @@ class Merchant < ApplicationRecord
   has_many :order_items, through: :items
   has_many :orders, through: :order_items
   has_many :users
+  has_many :discounts
```

Comment:
merchants have many discounts, discounts belong to merchants. should add a delete dependency